### PR TITLE
it-pushable: initial definition

### PR DESCRIPTION
**Pull request by yoursunny**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/it-pushable/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Type definitions for it-pushable 1.3
+// Project: https://github.com/alanshaw/it-pushable
+// Definitions by: Junxiao Shi <https://github.com/yoursunny>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.2
+
+/// <reference types="node" />
```
Comment (andrewbranch) [inline types/it-pushable/index.d.ts]: I’m probably overlooking something obvious, but why is this needed?

Comment (yoursunny) [reply types/it-pushable/index.d.ts]: It's needed for `AsyncIterable`. I'm going to change it to a "lib" reference.

File: types/it-pushable/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Type definitions for it-pushable 1.3
+// Project: https://github.com/alanshaw/it-pushable
+// Definitions by: Junxiao Shi <https://github.com/yoursunny>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.2
+
+/// <reference lib="esnext.asynciterable" />
```
Comment (andrewbranch) [inline types/it-pushable/index.d.ts]: Hm, so, I talked to the team about this, and I think we can’t have this here. It will pull those types in for anyone who installs these typings, even if their own lib settings don’t include it. (Granted, it seems unlikely that this is a real problem in practice, since anyone installing these typings _probably_ has a runtime that supports async iterable.) The right thing to do is probably to put this in the tsconfig `lib` section (for some reason I thought that DefinitelyTyped _only_ let you use `es6` and `dom`, but I might be misremembering), or, if that doesn’t work, go back to the `@types/node` reference since Node ships with an AsyncIterable implementation.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/it-pushable/index.d.ts (hunk 2 above)
Comment (yoursunny) [reply types/it-pushable/index.d.ts]: Moved to `tsconfig.json`. `npm test` passes.

**Comment by yoursunny** [comment]
@typescript-bot Can you tag this as "waiting for review"?
I already addressed the comments and need @andrewbranch 's review.

**Review by andrewbranch — APPROVED** [review]
